src/main/splitting/splitting.py

```diff
@@ -0,0 +1,106 @@
+import logging
+
+import pandas as pd
+
+from src.main.util import consts
+from src.main.preprocessing.code_tracker_handler import get_ct_language
+from src.main.splitting.tasks_tests_handler import check_tasks, create_in_and_out_dict
+from src.main.util.file_util import ct_file_condition, get_all_file_system_items, get_file_name_from_path,\
+    get_parent_folder_name, get_result_folder, write_based_on_language
+
+FRAGMENT = consts.CODE_TRACKER_COLUMN.FRAGMENT.value
+
+log = logging.getLogger(consts.LOGGER_NAME)
+
+
+def get_tasks_with_max_rate(tasks: list, test_results: list):
+    max_rate = max(test_results)
+    indices = [i for i, tr in enumerate(test_results) if tr == max_rate]
+    return [tasks[i] for i in indices], max_rate
+
+
+def check_tasks_on_correct_fragments(data: pd.DataFrame, tasks: list, in_and_out_files_dict: dict, file_log_info=""):
+    data[FRAGMENT] = data[FRAGMENT].fillna("")
+    # if run after preprocessing, this value can be taken from 'language' column
+    language = get_ct_language(data)
+    # add unique to logs
+    log.info(file_log_info + ", language is " + language + ", found " + str(data.shape[0]) + " fragments")
+
+    if language is not consts.LANGUAGE.NOT_DEFINED.value:
+        unique_fragments = list(data[FRAGMENT].unique())
+        log.info("Found " + str(len(unique_fragments)) + " unique fragments")
+
+        fragment_to_test_results_dict = dict(map(lambda f: (f, check_tasks(tasks, f, in_and_out_files_dict, language)), unique_fragments))
+        data[consts.CODE_TRACKER_COLUMN.TESTS_RESULTS.value] = data.apply(lambda row:
+                                                                          fragment_to_test_results_dict[row[FRAGMENT]], axis=1)
+    else:
+        data[consts.CODE_TRACKER_COLUMN.TESTS_RESULTS.value] = str([consts.TEST_RESULT.LANGUAGE_NOT_DEFINED.value] * len(tasks))
+
+    return language, data
+
+
+# since lists of tasks have small size, it should work faster than creating a set
+def intersect(list_1: list, list_2: list):
+    return [e for e in list_1 if e in list_2]
+
+
+# first version of splitting
+def find_real_splits(supposed_splits: list):
+    real_splits = []
+    if len(supposed_splits) == 0:
+        return real_splits
+
+    prev_split = supposed_splits[0]
+    prev_intersected_tasks = prev_split[consts.SPLIT_DICT.TASKS.value]
+
+    for curr_split in supposed_splits:
+        curr_intersected_tasks = intersect(prev_intersected_tasks, curr_split[consts.SPLIT_DICT.TASKS.value])
+        if len(curr_intersected_tasks) == 0:
+            # it means that the supposed task has changed, so we should split on prev_split
+            real_splits.append({consts.SPLIT_DICT.INDEX.value: prev_split[consts.SPLIT_DICT.INDEX.value],
```

Comment:
Can we move it in the separate function to add some values in the dictionary? Dictionary passed by links in python